Update to cardano-node 1.35-rc3

sevanspowell opened the pull request:
I have:

- Bumped the following dependencies to the `1.35.0-rc3` tag:
  - cardano-node
  - cardano-base
  - cardano-ledger
  - ouroboros-network
- Updated cardano-wallet code to build with `1.35.0-rc3` tag:
  - `evaluateTransactionExecutionUnits` has a new error type, updated
  `ErrAssignRedeemers` accordingly.
  - `SimpleScriptWitness` now takes a `SimpleScriptOrReferenceInput` instead of a
  `SimpleScript`, updated generators accordingly.
 
TODO:
- [x] Consider whether the `prop_balanceTransactionUnresolvedInputs` test makes sense anymore.
- [ ] Test the API output of `ErrAssignRedeemersTranslationError`.
- [x] Update readme compatibility matrix

### Issue Number

ADP-1907

sevanspowell committed:
Also bump cardano-nodes deps

- cardano-base
- cardano-ledger
- ouroboros-network

sevanspowell committed:
Update code to match rc3 changes

- 'evaluateTransactionExecutionUnits' has a new error type, update
  ErrAssignRedeemers accordingly.
- 'SimpleScriptWitness' now takes a 'SimpleScriptOrReferenceInput' instead of a
  'SimpleScript', update generators accordingly.

sevanspowell committed:
Initial bump to 1.35.0-rc3

- Bump cardano-wallet to 1.35.0-rc3

sevanspowell committed:
Fix unwrapping of TimeTranslationPastHorizon

Anviking commented on lib/core/src/Cardano/Wallet/Api/Server.hs:
```diff
@@ -5043,23 +5043,12 @@ instance IsServerError ErrAssignRedeemers where
             apiError err400 PastHorizon $ T.unwords
                 [ "The transaction's validity interval is past the horizon"
-                , "of safe slot-to-time conversions."
+                , "of safe slot-to-time conversions: " <> t <> "."
```
Ooh, nice! We have gotten some details back again. I think the exact `Text` comes from https://github.com/input-output-hk/cardano-wallet/blob/8e1b7239c090d3c110e8f24fb58396b5ec30327c/lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs#L1180 and is pretty long, so could we make this some `"\n\n Here are the full details:" <> t`  at the end?

sevanspowell replied:
Thank you. I was trying to write a test that would return this error (I was also worried about the readability), as chasing the definition of the error down proved more difficult than expected. But triggering the error also proved difficult. I will move it to the end :+1:

Anviking commented on lib/core/src/Cardano/Wallet/Api/Server.hs:
```diff
@@ -5043,23 +5043,12 @@ instance IsServerError ErrAssignRedeemers where
                 , "into a valid Plutus data:", pretty r <> "."
                 ]
-        ErrAssignRedeemersUnresolvedTxIns ins ->
-            -- Note that although this error is thrown from
-            -- '_assignScriptRedeemers', it's more related to balanceTransaction
-            -- in general than to assigning redeemers. Hence we don't mention
-            -- redeemers in the message.
-            apiError err400 UnresolvedInputs $ T.unwords
-                [ "The transaction I was given contains inputs I don't know"
-                , "about. Please ensure all foreign inputs are specified as "
-                , "part of the API request. The unknown inputs are:\n\n"
-                , pretty ins
-                ]
-        ErrAssignRedeemersTranslationError TimeTranslationPastHorizon ->
```
Probably `TranslationLogicMissingInput` is the replacement for the removed `UnknownTxIns` error, so:

Suggestion:
```haskell
        ErrAssignRedeemersTranslationError (TranslationLogicMissingInput inp) ->
             -- Note that although this error is thrown from
             -- '_assignScriptRedeemers', it's more related to balanceTransaction
             -- in general than to assigning redeemers. Hence we don't mention
             -- redeemers in the message.
             apiError err400 UnresolvedInputs $ T.unwords
                 [ "The transaction I was given contains inputs I don't know"
                 , "about. Please ensure all foreign inputs are specified as "
                 , "part of the API request. The unknown input is:\n\n"
                 , T.pack $ show inp
                 ]
```
then we should be able to keep the test (and this error code) unchanged. (Maybe it is in fact neater to more transparently only relying on `TranslationError`, but in-doubt, minimising the API changes for now seems safer to me — let me know if you disagree)

Anviking replied:
For some reason the test coverage seems very poor now. Not sure what's changed. I do get the new error, but just _very_ rarely. I think it might be best to mark it pending / remove, and if anything have a simple unit test.

I pushed to a separate branch, feel free to cherry-pick if you see fit
- a7a446c51e78a806e07b65e0306ab824f44ee934
- b4c2ec474e8803e7cbc58d0e46568fce627fe654

sevanspowell replied:
Thank you, wanted to ask you about this :+1:

Anviking committed:
Re-add UnresolvedInputs API error in balanceTx

Anviking committed:
Mark prop_balanceTransactionUnresolvedInputs pending

Anviking reviewed (approved):
As long as we preserve the `UnresolvedInputs` API error for good measure and update the readme I think we should merge!

sevanspowell committed:
Move full details of "time translation past horizon" error

- Move full details of "time translation past horizon" error to the end of the
  error message, in case the full details are long.

sevanspowell committed:
Update cardano-node in compatibility matrix

- Bump supported version of cardano-node in compatibility matrix to 1.35.0-rc3.
